main/src/ca/uwaterloo/flix/language/ast/OccurrenceAst.scala

```diff
@@ -139,61 +137,86 @@ object OccurrenceAst {
 
   case class MatchRule(pat: Pattern, guard: Option[Expr], exp: Expr)
 
-  case class TypeMatchRule(sym: Symbol.VarSym, tpe: Type, exp: Expr)
-
-  case class SelectChannelRule(sym: Symbol.VarSym, chan: Expr, exp: Expr)
-
-  case class ParYieldFragment(pat: Pattern, exp: Expr, loc: SourceLocation)
-
   /**
     * Represents occurrence information of binders, i.e., how a binder occurs in the program.
     * A binder may be a variable, function, or effect handler.
+    *
+    * The definitions form a lattice that allow for parallel evaluation:
+    *
+    * {{{
+    *       DontInline
+    *           |
+    *          Many
+    *           |
+    *       ManyBranch
+    *           |
+    * Once OnceInLambda OnceInLocalDef
+    *   \       |       /
+    *          Dead
```

Comment:
This is not entirely correct

Reply:
I'm not quite sure it is a lattice anymore... `ManyBranch` is not neccessarily the LUB of `Once` since it depends on whether or not two `Once` occurrences are in distinct branches or not... feels more like two distinct lattices.